File: languages/php/Bloock/Entity/Key/LocalCertificate.php
<?php

namespace Bloock\Entity\Key;

/**
 * Represents a local certificate along with its password.
 */
class LocalCertificate
{
    public array $pkcs12;
    public string $password;

    /**
     * Constructs a LocalCertificate object with the specified parameters.
     * @param array $pkcs12
     * @param string $password
     */
    public function __construct(array $pkcs12, string $password)
    {
        $this->pkcs12 = $pkcs12;
        $this->password = $password;
    }

    public static function fromProto(\Bloock\LocalCertificate $res): LocalCertificate
    {
        return new LocalCertificate(unpack('C*', $res->getPkcs12()), $res->getPassword());
    }

    public function toProto(): \Bloock\LocalCertificate
    {
        $p = new \Bloock\LocalCertificate();
        $p->setPkcs12(implode(array_map("chr", $this->pkcs12)));
        $p->setPassword($this->password);
        return $p;
    }
}

File: languages/python/bloock/entity/key/local_certificate.py
from __future__ import annotations

import bloock._bridge.proto.keys_entities_pb2 as proto


class LocalCertificate:
    """
    Represents a local certificate along with its password.
    """
    def __init__(
            self,
            pkcs12: bytes,
            password: str,
    ) -> None:
        """
        Constructs a LocalCertificate object with the specified parameters.
        :type password: object
        :type pkcs12: object
        :rtype: object
        """
        self.pkcs12 = pkcs12
        self.password = password

    @staticmethod
    def from_proto(certificate: proto.LocalCertificate) -> LocalCertificate:
        return LocalCertificate(
            pkcs12=certificate.pkcs12,
            password=certificate.password,
        )

    def to_proto(self) -> proto.LocalCertificate:
        return proto.LocalCertificate(
            pkcs12=self.pkcs12,
            password=self.password,
        )
